apps/openmw/mwmechanics/actors.cpp

```diff
@@ -488,33 +552,48 @@ namespace MWMechanics
 
         float x = fFatigueReturnBase + fFatigueReturnMult * endurance;
 
-        DynamicStat<float> fatigue = stats.getFatigue();
         fatigue.setCurrent (fatigue.getCurrent() + duration * x);
         stats.setFatigue (fatigue);
     }
 
-    class ExpiryVisitor : public EffectSourceVisitor
+    struct effectStruct
     {
-        private:
-            MWWorld::Ptr mActor;
-            float mDuration;
+        float mMagnitude;
+        float mRemainingTime;
+        MWMechanics::EffectKey mKey;
 
-        public:
-            ExpiryVisitor(const MWWorld::Ptr& actor, float duration)
-                : mActor(actor), mDuration(duration)
+        effectStruct(float magnitude, float remainingTime, MWMechanics::EffectKey key)
+            : mMagnitude(magnitude), mRemainingTime (remainingTime), mKey (key)
+        {
+        }
+
+        bool operator < (const effectStruct& str) const
+        {
+        return (mRemainingTime < str.mRemainingTime);
+        }
+
+    };
+
+    class SortByRemainingTimeVisitor : public EffectSourceVisitor
+    {
+            private:
+            bool dummy;
```

Comment:
I decided to change it back. The new implementation I had was messier in its own way. However, I didn't need to actually make a "dummy" variable, just pass in something as a parameter to the visitor, and the compiler accepted it.